src/utils/streams/array_stream.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Duplex } from 'stream';
+const NEWLINE_REGEX = /\r?\n/;
+
+export function map(mapfn) {
+  const stream = new Duplex({
+    write(chunk, _encoding, callback) {
+      // TS doesn't like `push` on this...
+      this.push(mapfn(chunk, callback));
+    },
+    read() {}
+  });
+
+  return stream;
+}
+
+export function split() {
```

Comment:
We also have https://github.com/elastic/kibana/blob/237590b63e036b5abdff812371f047e297238b34/src/utils/streams/split_stream.js, which may or may not be sufficient for our use case, but I'd prefer to update the one place that uses a split from `event-stream` to use the existing implementation.